packages/angular_devkit/core/src/jobs/api.ts

```diff
@@ -0,0 +1,476 @@
+}
+
+
+/**
+ * A Job instance, returned from scheduling a job. A Job instance is _not_ serializable.
+ */
+export interface Job<
+  ArgumentT extends JsonValue = JsonValue,
+  InputT extends JsonValue = JsonValue,
+  OutputT extends JsonValue = JsonValue,
+> {
+  /**
+   * Description of the job. Resolving the job's description can be done asynchronously, so this
+   * is an observable that will resolve when it's ready.
+   */
+  readonly description: Observable<JobDescription>;
+
+  /**
+   * Argument sent when scheduling the job. This is a copy of the argument.
+   */
+  readonly argument: ArgumentT;
+
+  /**
+   * The input to the job. This goes through the input channel as messages.
+   */
+  readonly input: Observer<InputT>;
+
+  /**
+   * Outputs of this job.
+   */
+  readonly output: Observable<OutputT>;
+
+  /**
+   * The current state of the job.
+   */
+  readonly state: JobState;
+
+  /**
+   * Get a channel that validates against the schema. Messages will be filtered by the schema.
+   * @param name The name of the channel.
+   * @param schema A schema to use to validate messages.
+   */
+  getChannel<T extends JsonValue>(name: string, schema?: schema.JsonSchema): Observable<T>;
+
+  /**
+   * Pings the job and wait for the resulting Pong before completing.
+   */
+  ping(): Observable<never>;
+
+  /**
+   * Stops the job from running. This is different than unsubscribing from the output as in it
+   * sends the JobInputKind.Stop raw input to the job.
+   */
+  stop(): void;
+
+  /**
+   * The JobInput events TO the job.
+   */
+  readonly rawInput: Observer<JobInput<InputT>>;
+
+  /**
+   * The JobEvent FROM the job.
+   */
+  readonly rawOutput: Observable<JobEvent<OutputT>>;
+}
+
+/**
+ * Options for scheduling jobs.
+ */
+export interface ScheduleJobOptions {
+  /**
+   * Where should logging be passed in. By default logging will be dropped.
+   */
+  logger?: LoggerApi;
+
+  /**
+   * Jobs that need to finish before scheduling this job. These dependencies will be passed
+   * to the job itself in its context.
+   */
+  dependencies?: Job | Job[];
+}
+
+export interface Registry<
+  MinimumArgumentValueT extends JsonValue = JsonValue,
+  MinimumInputValueT extends JsonValue = JsonValue,
+  MinimumOutputValueT extends JsonValue = JsonValue,
+> {
+  /**
+   * Get a job handler.
+   * @param name The name of the job to get a handler from.
+   */
+  get<
+    A extends MinimumArgumentValueT,
+    I extends MinimumInputValueT,
+    O extends MinimumOutputValueT,
+  >(name: JobName): Observable<JobHandler<A, I, O> | null>;
+}
+
+/**
+ * An interface that can schedule jobs.
+ */
+export interface Scheduler<
+  MinimumArgumentValueT extends JsonValue = JsonValue,
+  MinimumInputValueT extends JsonValue = JsonValue,
+  MinimumOutputValueT extends JsonValue = JsonValue,
+> {
+  /**
+   * Get a job description for a named job.
+   *
+   * @param name The name of the job.
+   * @returns A description, or null if the job cannot be scheduled.
```

Comment:
This will return `null` if the job was not scheduled or cannot be scheduled? Does a job need to be scheduled to get its description?

Reply:
This will return `null` if the scheduler does not have a description available (so that `scheudle()` would error out). Clarified.